app/src/main/java/com/example/recipeapp/viewmodels/LoginViewModel.java

```diff
@@ -27,11 +28,12 @@ public void login(View view) {
                 username,
                 password,
                 (user, e) -> {
+                    Context context = view.getContext();
                     if(e != null) {
                         Log.i(TAG, "Issue with login" + e);
+                        Toast.makeText(context, "Unable to login", Toast.LENGTH_SHORT).show();
```

Comment:
View Model should not hold any UI logic, it should be done in Activity.

- View Model just holds backend logic,